packages/server/src/router.ts

```diff
@@ -609,6 +671,27 @@ export class Router<
   }
 }
 
-export function router<TContext = void>() {
-  return new Router<TContext, {}, {}, {}, DefaultErrorShape>();
+/** Subclass of `VNextRouter` with `TInputContext` and `TContext` set to the same type, for backcompat. */
+export class LegacyRouter<
```

Comment:
```suggestion
/** 
 * Subclass of `Router` with `TInputContext` and `TContext` set to the same type, for backcompat. 
 * @deprecated
 */
export class LegacyRouter<
```

Reply:
Only reason I didn't deprecate it yet is in case we want to switch to an options bag for typeargs before the next major. If that's the plan then it'd be not great to tell people that this one is deprecated and they should switch to something that will change again. But maybe both `Router` and `LegacyRouter` could be marked as `@internal` so that anyone using them in future does so at their own risk, what do you think?

Reply:
I think it should be marked as internal in general -- I don't want people using `Router`, but it's ok if they use `AnyRouter` or `inferProcedureOutput<X>`, etc that we can maintain

Reply:
Actually, on further thought, you're right. I was thinking the `@deprecated` tag would point to the new `Router`, but how about we deprecate it with no replacement. I don't know of any use case for an external user using the `Router` class, with all its typeargs directly, and it's pretty iffy to have to support unknown use cases. I'd vote to make it not part of the API surface at all in the next major. If someone has a use-case they can raise an issue and maybe some type helper can be exposed.

But having it part of the API surface reduced the ability to refactor easily, and having so many typeargs means it makes more sense as internal generally.

So basically exactly what you suggested!